README.md

```diff
@@ -1,47 +1,271 @@
+which stores how far through the `state_groups` table the compressor has scanned.
+
+The tool can be run manually when you are running out of space, or be scheduled to run
+periodically.
+
+The output from the auto_compressor will be sent to `auto_compressor.log` (in the directory
+that the compressor is run from).
+
+## Building 
+
+This tool requires `cargo` to be installed. See https://www.rust-lang.org/tools/install
+for instructions on how to do this.
+
+To build `auto_compressor`, clone this repository and navigate to the `autocompressor/` 
+subdirectory. Then execute `cargo build`.
+
+This will create an executable and store it in `auto_compressor/target/debug/auto_compressor`.
+
+## Example usage
+```
+$ auto_compressor -p postgresql://user:pass@localhost/synapse -c 500 -l '100,50,25' -n 100
+```
+## Running Options
+
+- -p [POSTGRES_LOCATION] **Required**  
+The configuration for connecting to the Postgres database. This should be of the form
+`"postgresql://username:password@mydomain.com/database"` or a key-value pair
+string: `"user=username password=password dbname=database host=mydomain.com"`
+See https://docs.rs/tokio-postgres/0.7.2/tokio_postgres/config/struct.Config.html
+for the full details.
+
+- -c [CHUNK_SIZE] **Required**  
+The number of state groups to work on at once. All of the entries from state_groups_state are
+requested from the database for state groups that are worked on. Therefore small chunk
+sizes may be needed on machines with low memory. Note: if the compressor fails to find
+space savings on the chunk as a whole (which may well happen in rooms with lots of backfill
+in) then the entire chunk is skipped.
+
+- -n [CHUNKS_TO_COMPRESS] **Required**  
+*CHUNKS_TO_COMPRESS* chunks of size *CHUNK_SIZE* will be compressed.
+
+- -d [LEVELS]  
+Sizes of each new level in the compression algorithm, as a comma-separated list.
+The first entry in the list is for the lowest, most granular level, with each
+subsequent entry being for the next highest level. The number of entries in the
+list determines the number of levels that will be used. The sum of the sizes of
+the levels affects the performance of fetching the state from the database, as the
+sum of the sizes is the upper bound on the number of iterations needed to fetch a
+given set of state. [defaults to "100,50,25"]
+
+## Scheduling the compressor
+Create the following script and save it somewhere sensible
+(e.g. `/home/synapse/compress.sh`)
+
+```
+#!/bin/bash
+
+cd /home/synapse/rust-synapse-compress-state/
+
+URL=postgresql://user::pass@domain.com/synapse
+CHUNK_SIZE=500
+LEVELS="100,50,25"
+NUMBER_OF_CHUNKS=100
+
+/home/synapse/rust-synapse-compress-state/target/debug/auto_compressor \
+-p $URL \
+-c $CHUNK_SIZE \
+-l $LEVELS \
+-n $NUMBER_OF_ROOMS
+```
+
+Make it executable with `chmod +x compress.sh`
+
+Then run `crontab -e` to edit your scheduled tasks and add the following:
+
+```
+# Run every day at 3:00am
+00 3 * * * /home/synapse/compress.sh
+```
```

Comment:
I think its probably best to skip this, there are a million and one ways of doing it and sysadmins should know.